discord/role.py

```diff
@@ -79,70 +183,104 @@ class RoleTags:
-
+        self._premium_subscriber: bool | None = _parse_tag_bool(
+            data, "premium_subscriber"
+        )
+        self._available_for_purchase: bool | None = _parse_tag_bool(
+            data, "available_for_purchase"
+        )
+        # here discord did things in a normal and logical way for once
+        self._is_guild_product_role: bool | None = data.get("is_guild_product_role")
+
+    @cached_slot_property("_type")
+    def type(self) -> RoleType:
+        """Determine the role type based on tag flags."""
+        # Bot role
+        if self.bot_id is not None:
+            return RoleType.APPLICATION
+
+        # Role connection
+        if self._guild_connections is True:
+            return RoleType.CONNECTION
+
+        # Paid roles
+        if self._is_guild_product_role is True:
+            return RoleType.GUILD_PRODUCT
+
+        # Booster role
+        if self._premium_subscriber is True:
+            return RoleType.BOOSTER
+
+        # subscription roles
+        if (
+            self.integration_id is not None
+            and self._premium_subscriber is None
+            and self.subscription_listing_id is not None
+        ):
+            if self._available_for_purchase is True:
+                return RoleType.PREMIUM_SUBSCRIPTION_TIER
+            return RoleType.DRAFT_PREMIUM_SUBSCRIPTION_TIER
+
+        # integration role (Twitch/YouTube)
+        if self.integration_id is not None:
+            return RoleType.INTEGRATION
+
+        # Seeing how messed up this is it wouldn't be a surprise if this happened
+        return RoleType.UNKNOWN
+
+    @deprecated("RoleTags.type", "2.7")
     def is_bot_managed(self) -> bool:
         """Whether the role is associated with a bot."""
         return self.bot_id is not None
 
+    @deprecated("RoleTags.type", "2.7")
     def is_premium_subscriber(self) -> bool:
         """Whether the role is the premium subscriber, AKA "boost", role for the guild."""
         return self._premium_subscriber is None
 
+    @deprecated("RoleTags.type", "2.7")
     def is_integration(self) -> bool:
         """Whether the guild manages the role through some form of
         integrations such as Twitch or through guild subscriptions.
         """
         return self.integration_id is not None
 
+    @deprecated("RoleTags.type", "2.7")
     def is_available_for_purchase(self) -> bool:
-        """Whether the role is available for purchase.
-
-        Returns ``True`` if the role is available for purchase, and
-        ``False`` if it is not available for purchase or if the role
-        is not linked to a guild subscription.
-
-        .. versionadded:: 2.7
-        """
-        return self._available_for_purchase is None
+        """Whether the role is available for purchase."""
+        return self._available_for_purchase is True
 
+    @deprecated("RoleTags.type", "2.7")
     def is_guild_connections_role(self) -> bool:
-        """Whether the role is a guild connections role.
-
```

Comment:
Wait this could be removed if this pr is merged before 2.7 since it was added after 2.6.1. Sane for `is_available_for_purchase`